services/ad-tech/src/routes/common-router.ts

```diff
@@ -0,0 +1,159 @@
+/*
+ Copyright 2022 Google LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+      https://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import express, {NextFunction, Request, Response} from 'express';
+
+import {
+  handleAttributionSourceRegistration,
+  handleAttributionTriggerRegistration,
+} from '../lib/attribution-reporting-helper.js';
+import {
+  Report,
+  ReportCategory,
+  ReportStore,
+} from '../controllers/report-store.js';
+import {
+  getAdTemplateVariables,
+  getTemplateVariables,
+} from '../lib/template-utils.js';
+
+export const CommonRouter = express.Router();
+
+// ************************************************************************
+// HTTP Header Configurations
+// ************************************************************************
+CommonRouter.use((req: Request, res: Response, next: NextFunction) => {
+  // Explicitly allow loading in fenced-frame.
+  if (req.get('Sec-Fetch-Dest') === 'fencedframe') {
+    res.setHeader('Supports-Loading-Mode', 'fenced-frame');
+  }
+  // Enable debug reports for Atribution Reporting.
+  res.cookie('ar_debug', '1', {
+    sameSite: 'none',
+    secure: true,
+    httpOnly: true,
+  });
+  // Enable CORS.
+  if (req.headers.origin?.startsWith('https://privacy-sandbox-demos-')) {
+    res.setHeader('Access-Control-Allow-Origin', req.headers.origin!);
+    res.setHeader('Access-Control-Allow-Credentials', 'true');
+    res.setHeader('Access-Control-Private-Network', 'true');
+  }
+  // Observe browsing topics.
+  res.setHeader('Observe-Browsing-Topics', '?1');
```

Comment:
these Topics API related integrations to the router should be in a separate topics-router.js even if they're common.